File: 27_feb/change/change.py
def make_change(value):
	curr = [25, 10, 5, 1]
	change = {}
	for coin in curr:
		diff = value // coin
		if diff >= 1:
			if coin == 25:
				change['quarter'] = diff
			elif coin == 10:
				change['dime'] = diff
			elif coin == 5:
				change['nickel'] = diff
			else:
				change['penny'] = diff
			value -= diff * coin
	return change

print make_change(94)

File: 27_feb/change/change.java
import java.util.LinkedHashMap;
import static java.lang.System.out;

class change {
	private static LinkedHashMap<String, Integer> curr = new LinkedHashMap<String, Integer>();

	private static LinkedHashMap<String, Integer> makeChange(int value) {
		LinkedHashMap<String, Integer> change = new LinkedHashMap<String, Integer>();
		for (String key: curr.keySet()) {
			if (value <= 0) break;
			int diff = value / curr.get(key);
			if (diff >= 1) {
				change.put(key, diff);
				value -= curr.get(key) * diff;
			}
		}
		return change;
	}

	public static void main(String[] args) {
		curr.put("quarter", 25);
		curr.put("dime", 10);
		curr.put("nickel", 5);
		curr.put("penny", 1);
		out.println(makeChange(94));
	}
}
